### PR TITLE
Rename TrustyAI `apiGroup` to `trustyai.opendatahub.io`

### PR DESCRIPTION
**Pull request by ruivieira**
Currently the `apiVersion` for TrustyAI resources is `trustyai.opendatahub.io.trustyai.opendatahub.io/v1alpha1`.

This name does not follow best practices and should be changed to the simpler `trustyai.opendatahub.io/v1alpha1`.

Although no functional change will take place, this renaming will impact:

- Documentation
- Examples will need to be updated
- CI

and merge should be on hold until the above changes are prepared.

_Note_: CI will fail for this PR until it is updated.

### CODE REVIEW

**Comment by spolti** [comment]
I guess it needs to be renamed as well:

https://github.com/trustyai-explainability/trustyai-service-operator/blob/main/PROJECT

domain, if I not mistaken is `opendatahub.io` and group `trustyai`

**Comment by spolti** [comment]
hi @ruivieira, how did you updated the apiversion?
The operator-sdk and controller-gen should be able to update anything that might be needed.

btw, what is the operator-sdk version in use?
Didn't find any information about it.

**Comment by ruivieira** [comment]
> hi @ruivieira, how did you updated the apiversion? The operator-sdk and controller-gen should be able to update anything that might be needed.
> 
> btw, what is the operator-sdk version in use? Didn't find any information about it.

Hi @spolti, the apiversion was updated by:

- Updating `api/v1alpha1/groupversion_info.go`
- Using Kubebuilder's `make manifests`, `make generate`
- But some apiVersions had to be manually changed
- operator-sdk is v1.32.0

File: PROJECT [hunk 1]
```diff
@@ -2,7 +2,7 @@
 # This file is used to track the info used to scaffold your project
 # and allow the plugins properly work.
 # More info: https://book.kubebuilder.io/reference/project-config.html
-domain: trustyai.opendatahub.io
```
Comment (danielezonca) [inline PROJECT:5]: Are these changes necessary?

File: PROJECT [hunk 2]
```diff
@@ -15,8 +15,8 @@ resources:
     crdVersion: v1
     namespaced: true
   controller: true
-  domain: trustyai.opendatahub.io
-  group: trustyai.opendatahub.io
```
Comment (danielezonca) [inline PROJECT:19]: Are these changes necessary?
